File: myFinance_improved_V2/lib/features/sales_invoice/domain/entities/cash_location.dart
import 'package:equatable/equatable.dart';

/// Cash location entity for payment method
class CashLocation extends Equatable {
  final String id;
  final String name;
  final String type; // 'bank' or 'cash'
  final String storeId;
  final bool isCompanyWide;
  final String currencyCode;
  final String? bankAccount;
  final String? bankName;

  const CashLocation({
    required this.id,
    required this.name,
    required this.type,
    required this.storeId,
    required this.isCompanyWide,
    required this.currencyCode,
    this.bankAccount,
    this.bankName,
  });

  /// Helper getters for UI display
  String get displayName {
    if (type == 'bank' && bankName != null) {
      return '$name - $bankName';
    }
    return name;
  }

  String get displayType {
    switch (type) {
      case 'bank':
        return 'Bank';
      case 'cash':
        return 'Cash';
      case 'vault':
        return 'Vault';
      default:
        // Capitalize first letter for unknown types
        return type.isNotEmpty
            ? '${type[0].toUpperCase()}${type.substring(1)}'
            : 'Unknown';
    }
  }

  bool get isBank => type == 'bank';
  bool get isCash => type == 'cash';
  bool get isVault => type == 'vault';

  @override
  List<Object?> get props => [
        id,
        name,
        type,
        storeId,
        isCompanyWide,
        currencyCode,
        bankAccount,
        bankName,
      ];
}

File: website/src/features/sale-product/domain/entities/CashLocation.ts
/**
 * CashLocation Entity
 * Domain entity representing a cash/bank/vault location
 */

export type CashLocationType = 'cash' | 'bank' | 'vault';

export class CashLocation {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly type: CashLocationType,
    public readonly storeId: string | null,
    public readonly isCompanyWide: boolean,
    public readonly currencyCode: string | null,
    public readonly bankAccount: string | null,
    public readonly bankName: string | null
  ) {}

  get displayName(): string {
    return this.name;
  }

  get typeLabel(): string {
    return this.type;
  }

  isAvailableForStore(storeId: string): boolean {
    return this.isCompanyWide || this.storeId === storeId;
  }

  static create(data: {
    id: string;
    name: string;
    type: CashLocationType;
    storeId: string | null;
    isCompanyWide: boolean;
    currencyCode: string | null;
    bankAccount: string | null;
    bankName: string | null;
  }): CashLocation {
    return new CashLocation(
      data.id,
      data.name,
      data.type,
      data.storeId,
      data.isCompanyWide,
      data.currencyCode,
      data.bankAccount,
      data.bankName
    );
  }
}
